spring-shell-core/src/main/java/org/springframework/shell/core/command/Command.java

```diff
@@ -74,4 +76,70 @@ default List<String> getAliases() {
 	 */
 	ExitStatus execute(CommandContext commandContext) throws Exception;
 
+	/**
+	 * Creates and returns a new instance of a {@code Builder} for defining and
+	 * constructing commands.
+	 * <p>
+	 * The builder allows customization of command properties such as name, description,
+	 * group, help text, aliases, and execution logic.
+	 * @return a new {@code Builder} instance for configuring and creating commands
+	 */
+	static Builder builder() {
+		return new DefaultCommandBuilder();
+	}
+
+	/**
+	 * Builder for creating command.
+	 */
+	interface Builder {
+
+		/**
+		 * Set the name of the command.
+		 * @return this builder
+		 */
+		Builder name(String name);
+
+		/**
+		 * Set the description of the command.
+		 * @return this builder
+		 */
+		Builder description(String description);
+
+		/**
+		 * Set the help of the command.
+		 * @return this builder
+		 */
+		Builder help(String help);
+
+		/**
+		 * Set the group of the command.
+		 * @return this builder
+		 */
+		Builder group(String group);
+
+		/**
+		 * Set the aliases of the command.
+		 * @return this builder
+		 */
+		Builder aliases(String... aliases);
+
+		/**
+		 * Set the aliases of the command.
+		 * @return this builder
+		 */
+		Builder aliases(List<String> aliases);
+
+		/**
+		 * Set command execution logic.
+		 * @return this builder
+		 */
+		Builder execute(Consumer<CommandContext> commandExecutor);
+
+		/**
+		 * Build the {@link AbstractCommand}.
+		 */
+		AbstractCommand build();
```

Comment:
`AbstractCommand` is a leaky abstraction here (look at the import coming from a different package). The builder should return the interface type (or at least a package private type that we can't break from the outside, but since we don't have that, returning the interface type is the way to go).